support/hololens/ServoApp/BrowserPage.cpp

```diff
@@ -31,15 +33,13 @@ void BrowserPage::Shutdown() {
   log("BrowserPage::Shutdown()");
 
   if (mServo != nullptr) {
-    if (!IsLoopRunning()) {
+    if (!mLooping) {
       // FIXME: this should not happen. In that case, we can't send the
       // shutdown event to Servo.
     } else {
-      HANDLE hEvent = ::CreateEventA(nullptr, FALSE, FALSE, sShutdownEvent);
-      SendEventToServo({{Event::SHUTDOWN}});
-      log("Waiting for Servo to shutdown");
-      ::WaitForSingleObject(hEvent, INFINITE);
-      StopRenderLoop();
+      RunOnGLThread([=] { mServo->RequestShutdown(); });
+      mLoopTask->wait();
```

Comment:
Is this basically going to wait on `cancel_current_task()` to be called from inside the loop? 

Since what will eventually happen as a result of the `RequestShutdown` call, is that `OnShutdownComplete` will be called and will stop the loop. 

And `OnShutdownComplete` will not signal back here, so if this is will be notified once the loop stops, it looks right, otherwise you might to keep the `sShutdownEvent`, and dispatch it when the loop stops.

Reply:
> Is this basically going to wait on cancel_current_task() to be called from inside the loop?

Yes.

`sShutdownEvent` is not necessary. Wait() will block until the loop stops, and will stop only if `OnShutdownComplete` is called.

Reply:
ok